lightkurve/targetpixelfile.py

```diff
@@ -537,3 +543,195 @@ def get_bkg_lightcurve(self, aperture_mask=None):
         aperture_mask = self._parse_aperture_mask(aperture_mask)
         return LightCurve(flux=np.nansum(self.flux_bkg[:, aperture_mask], axis=1),
                           time=self.time, flux_err=self.flux_bkg_err)
+
+
+class TargetPixelFileFactory(object):
+
+    def __init__(self, n_cadences, n_rows, n_cols, target_id="unnamed-target",
+                 keywords={}):
+        self.n_cadences = n_cadences
+        self.n_rows = n_rows
+        self.n_cols = n_cols
+        self.target_id = target_id
+        self.keywords = keywords
+
+        # Initialize the 3D data structures
+        self.raw_cnts = np.empty((n_cadences, n_rows, n_cols), dtype='int')
+        self.flux = np.empty((n_cadences, n_rows, n_cols), dtype='float32')
+        self.flux_err = np.empty((n_cadences, n_rows, n_cols), dtype='float32')
+        self.flux_bkg = np.empty((n_cadences, n_rows, n_cols), dtype='float32')
+        self.flux_bkg_err = np.empty((n_cadences, n_rows, n_cols), dtype='float32')
+        self.cosmic_rays = np.empty((n_cadences, n_rows, n_cols), dtype='float32')
+
+        # Set 3D data defaults
+        self.raw_cnts[:, :, :] = -1
+        self.flux[:, :, :] = np.nan
+        self.flux_err[:, :, :] = np.nan
+        self.flux_bkg[:, :, :] = np.nan
+        self.flux_bkg_err[:, :, :] = np.nan
+        self.cosmic_rays[:, :, :] = np.nan
+
+        # Initialize the 1D data structures
+        self.mjd = np.zeros((n_cadences), dtype='float64')
+        self.time = np.zeros((n_cadences), dtype='float64')
+        self.timecorr = np.zeros((n_cadences), dtype='float32')
+        self.cadenceno = np.zeros((n_cadences), dtype='int')
+        self.quality = np.zeros((n_cadences), dtype='int')
+        self.pos_corr1 = np.zeros((n_cadences), dtype='float32')
+        self.pos_corr2 = np.zeros((n_cadences), dtype='float32')
+
+    def add_cadence(self, n, raw_cnts=None, flux=None, flux_err=None,
+                    flux_bkg=None, flux_bkg_err=None, cosmic_rays=None,
+                    mjd=None, time=None, timecorr=None, cadenceno=None,
+                    quality=None, pos_corr1=None, pos_corr2=None):
+        fields = ['raw_cnts', 'flux', 'flux_err', 'flux_bkg', 'flux_bkg_err',
+                  'cosmic_rays', 'mjd', 'time', 'timecorr', 'cadenceno',
+                  'quality', 'pos_corr1', 'pos_corr2']
+        for field in fields:
+            if locals()[field] is not None:
+                vars(self)[field][n] = locals()[field]
+
+    def get_tpf_object(self):
+        return KeplerTargetPixelFile(self._make_hdulist())
+
+    def write_fits(self, output_fn=None, overwrite=False):
```

Comment:
Maybe output_fn should have a sensible default? out.fits, or something yet more sensible?